### PR TITLE
scheduler, operator:  support operatorLimitCounter for region-schedule-limit

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

ref https://github.com/tikv/pd/issues/3350

### What is changed and how it works?

Add `OperatorLimitCounter` for region-schedule-limit. For the other limit situations, I will add support those in the next several pr.

image

image

### Release note

* No release note

### CODE REVIEW

File: server/schedule/operator/metrics.go [hunk 1]
```diff
@@ -24,8 +24,18 @@ var (
...
+			Namespace: "pd",
+			Subsystem: "schedule",
+			Name:      "operator_limit",
+			Help:      "Counter of operator meeting limit",
+		}, []string{"type", "name"})
 )
 
 func init() {
 	prometheus.MustRegister(operatorStepDuration)
+	prometheus.MustRegister(OperatorLimitCounter)
```
Comment (lhy1024) [inline server/schedule/operator/metrics.go:40]: Maybe it will be more convenient if support the source store and target store.

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM. it will be helpful!

File: server/schedule/operator/metrics.go (hunk 1 above)
Comment (Yisaer) [reply server/schedule/operator/metrics.go:40]: The labels for recording `source` and `target` needs to be tested considering the performance and storage for prometheus as same as https://github.com/tikv/pd/pull/3320

File: server/schedulers/balance_region.go [hunk 2]
```diff
@@ -126,7 +126,11 @@ func (s *balanceRegionScheduler) EncodeConfig() ([]byte, error) {
 }
 
 func (s *balanceRegionScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
-	return s.opController.OperatorCount(operator.OpRegion)-s.opController.OperatorCount(operator.OpMerge) < cluster.GetOpts().GetRegionScheduleLimit()
+	allowed := s.opController.OperatorCount(operator.OpRegion)-s.opController.OperatorCount(operator.OpMerge) < cluster.GetOpts().GetRegionScheduleLimit()
+	if !allowed {
+		operator.OperatorLimitCounter.WithLabelValues(s.GetType(), operator.OpRegion.String()).Inc()
```
Comment (rleungx) [inline server/schedulers/balance_region.go:131]: BTW, I think we can only record if scheduler can run in `scheduler is running` monitor and use this metrics to record the limit separately.

Comment (Yisaer) [reply server/schedulers/balance_region.go:131]: updated.

File: server/schedulers/scatter_range.go [hunk 3]
```diff
@@ -193,7 +193,11 @@ func (l *scatterRangeScheduler) EncodeConfig() ([]byte, error) {
 }
 
 func (l *scatterRangeScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
-	return l.OpController.OperatorCount(operator.OpRange) < cluster.GetOpts().GetRegionScheduleLimit()
+	allowed := l.OpController.OperatorCount(operator.OpRange) < cluster.GetOpts().GetRegionScheduleLimit()
+	if !allowed {
+		operator.OperatorLimitCounter.WithLabelValues(l.GetType(), operator.OpRegion.String()).Inc()
```
Comment (rleungx) [inline server/schedulers/scatter_range.go:198]: There is problem here, but not related to this PR. See https://github.com/tikv/pd/issues/3356

File: metrics/grafana/pd.json [hunk 4]
```diff
@@ -2170,6 +2170,96 @@
...
+          "seriesOverrides": [],
+          "spaceLength": 10,
+          "stack": false,
+          "steppedLine": false,
+          "targets": [
+            {
+              "expr": "sum(delta(pd_schedule_operator_limit{instance=\"$instance\"}[1m])) by (type,name)",
+              "format": "time_series",
+              "intervalFactor": 2,
+              "legendFormat": "{{type}}-{{name}}-limit",
```
Comment (rleungx) [inline metrics/grafana/pd.json:2219]:
```suggestion
              "legendFormat": "{{type}}-{{name}}",
```

File: server/schedulers/scatter_range.go (hunk 3 above)
Comment (Yisaer) [reply server/schedulers/scatter_range.go:198]: cuz the problem is tracked by #3356, I resolved this conversation.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**